### ISSUE TITLE
SpatialOverlay - splitting GeopaparazziOverlay

### ISSUE THREAD

**Issue by mj10777** [geopaparazzi/geopaparazzi#192]
Last May I had experimented with splitting GeopaparazziOverlay so that:
- library specific portions could be take out of the Application code
  - mainly the geometry overly logic

It was very simple to split and worked well, but to be frank I forgot about it completely until now
- compiling the `nf_editing` complained probably because of the re-fracturing
  - so I remove the directory
    `geopaparazzimapsforge/src/eu/geopaparazzi/mapsforge/mapsdirmanager/maps/overlay/`

and everything compiled correctly.

Still I think it would be a good idea to have this in the library
- a the default displaying logic for geometries
  - an another application using the library could then just override or replace if it wanted to do something else

If you think this is a good idea, I will send you an archiv of that directory.

I was also hoping that the gps tracing/logging could be done also
- but that was so intertwined with application code

**Comment by mj10777** [comment]
I believe it would be a good idea to split the geometry specific Overlay logic out of the general Overlay code.

**Comment by moovida** [comment]
I agree with you but the effort is not few, so I am leaving this open in the hope to have resources for this at some point.
